src/main.rs

```diff
@@ -535,92 +550,77 @@ impl ManParser for Type1 {
         manpage.contains(r#".SH "OPTIONS""#)
     }
 
-    fn parse_man_page(&mut self, manpage: &str) -> Option<String> {
-        unimplemented!();
+    fn parse_man_page(&self, manpage: &str, cmdname: &str) -> Option<String> {
+        let options_section_re = regex!(r#"\.SH "OPTIONS"((?s:.)*?)(\.SH|\Z)"#);
+        let options_section_matched = options_section_re.find(manpage);
+        let mut options_section = options_section_matched.unwrap().as_str();
+
+        let options_parts_re = regex!(r"\.PP((?s:.)*?)\.RE");
+        let mut options_matched = options_parts_re.captures(options_section);
+        // add_diagnostic(format!("Command is {}", cmdname));
+
+        if options_matched.is_none() {
+            // add_diagnostic("Unable to find options");
+            return self
+                .fallback(options_section, cmdname)
+                .or_else(|| self.fallback2(options_section, cmdname));
+        }
+
+        let mut completions = Completions::new(cmdname);
+        while let Some(mat) = options_matched {
+            let mut data = mat.get(1).unwrap().as_str();
+            let last_dotpp_index = data.rfind(".PP");
+            if let Some(idx) = last_dotpp_index {
+                data = &data[idx + 3..];
+            }
+
+            let data = remove_groff_formatting(data);
+            if let Some(data) = data.splitn(2, ".RS 4").next_tuple::<(_, _)>() {
+                let option_name = data.0.trim();
+                if option_name.contains('-') {
+                    let option_name = unquote_double_quotes(option_name);
+                    let option_name = unquote_single_quotes(option_name);
+                    let option_desc = data.1.trim().replace('\n', " ");
+                    completions.add(option_name, &option_desc);
```

Comment:
Could eagerly extract tuple contents so reader has one fewer intermediate binding to remember.
```suggestion
            if let Some((option_name, option_desc)) = data.splitn(2, ".RS 4").next_tuple::<(_, _)>() {
                let option_name = option_name.trim();
                if option_name.contains('-') {
                    let option_name = unquote_double_quotes(option_name);
                    let option_name = unquote_single_quotes(option_name);
                    let option_desc = option_desc.trim().replace('\n', " ");
                    completions.add(option_name, &option_desc);
```